include/seqan/modifier/modifier_iterator.h

```diff
@@ -538,11 +530,20 @@ position(ModifiedIterator<THost, TSpec> const & me)
     return position(host(me));
 }
 
+// rooted overload
+template <typename TContainer1, typename TIterator, typename TSpec1, typename TSpec2, typename TContainer2>
+inline typename Position<ModifiedIterator<Iter<TContainer1, AdaptorIterator<TIterator, TSpec1> >, TSpec2> const>::Type
+position(ModifiedIterator<Iter<TContainer1, AdaptorIterator<TIterator, TSpec1> >, TSpec2> const & me,
+        TContainer2 const &)
+{
+    return position(me); // rooted has container
+}
+
 template <typename THost, typename TSpec, typename TContainer>
 inline typename Position<ModifiedIterator<THost, TSpec> const>::Type
 position(ModifiedIterator<THost, TSpec> const & me, TContainer const &cont)
 {
-    return position(host(me), cont);
+    return me - begin(cont, Standard());
```

Comment:
Sure there is a generic function, but it doesn't work ;) 
... Because the host iterator might be on a completely different alphabet then the container (and the mod iterator).